acceptance/analyzer_test.go

```diff
@@ -145,34 +121,57 @@ func setupAnalyzeFixtures(t *testing.T) analyzeFixtures {
 
 	// Registry
 
+	// setup read and write images
 	someAppName := "some-app-image-" + h.RandString(10)
 	fixtures.authRegAppImage, fixtures.authRegAuthConfig = buildAuthRegistryImage(
 		t,
 		someAppName,
 		filepath.Join("testdata", "analyzer", "app-image"),
+		testRegistry,
 		"--build-arg", "fromImage="+containerBaseImage,
 		"--build-arg", "metadata="+appMeta,
 	)
-	fixtures.readOnlyRegAppImage = readOnlyRegistry.RepoName(someAppName)
+	testRegistry.Add(someAppName, ih.NewImagePrivileges(ih.Readable, ih.Writable))
 
 	someOtherAppName := "some-other-app-image-" + h.RandString(10)
 	fixtures.authRegOtherAppImage, _ = buildAuthRegistryImage(
 		t,
 		someOtherAppName,
 		filepath.Join("testdata", "analyzer", "app-image"),
+		testRegistry,
 		"--build-arg", "fromImage="+containerBaseImage,
 		"--build-arg", "metadata="+appMeta,
 	)
+	testRegistry.Add(someOtherAppName, ih.NewImagePrivileges(ih.Readable, ih.Writable))
+
+	// setup read only images
+	someReadOnlyAppName := "some-readonly-app-image-" + h.RandString(10)
+	fixtures.readOnlyRegAppImage, _ = buildAuthRegistryImage(
+		t,
+		someReadOnlyAppName,
+		filepath.Join("testdata", "analyzer", "app-image"),
+		testRegistry,
+		"--build-arg", "fromImage="+containerBaseImage,
+		"--build-arg", "metadata="+appMeta,
+	)
+	testRegistry.Add(someReadOnlyAppName, ih.NewImagePrivileges(ih.Readable))
 
 	someCacheName := "some-cache-image-" + h.RandString(10)
 	fixtures.authRegCacheImage, _ = buildAuthRegistryImage(
 		t,
 		someCacheName,
 		filepath.Join("testdata", "analyzer", "cache-image"),
+		testRegistry,
 		"--build-arg", "fromImage="+containerBaseImage,
 		"--build-arg", "metadata="+cacheMeta,
 	)
-	fixtures.readOnlyRegCacheImage = readOnlyRegistry.RepoName(someCacheName)
+	testRegistry.Add(someCacheName, ih.NewImagePrivileges(ih.Readable, ih.Writable))
+	fixtures.readOnlyRegCacheImage = testRegistry.RepoName(someCacheName)
```

Comment:
It seems that this image has read & write permission... should we make another image for `readOnlyRegCacheImage`?

Reply:
check the following [commit](https://github.com/buildpacks/lifecycle/pull/685/commits/59c111c24edd1c3bb2cdcaba6ba521a5b3718d4d)